new_platforms/scripts/VirtualMachineSgxSettings.psm1

```diff
@@ -0,0 +1,283 @@
+}
+
+### -----------------------------------
+### Helper Functions
+### -----------------------------------
+
+Function Get-VMSgxMemorySettingData([Microsoft.HyperV.PowerShell.VirtualMachine] $Vm)
+{
+    $Mem =  Get-WmiObject -Namespace root/virtualization/v2 -Class Msvm_MemorySettingData |
+        Where-Object { $_.InstanceId -eq "Microsoft:$($Vm.Id)\$VM_MEMORY_SETTINGS_GUID" }
+
+    If (!$Mem)
+    {
+        Write-Error ($Strings.ERR_NO_MEM_SETTING_DATA_OBJECT -f $Vm.Name)
+    }
+
+    Return $Mem
+}
+
+Function Get-VMSgxManagementService()
+{
+    $Svc = Get-WmiObject -Namespace root/virtualization/v2 -Class Msvm_VirtualSystemManagementService |
+        Where-Object {
+            ($_.CreationClassName -eq "Msvm_VirtualSystemManagementService") -and
+            ($_.Name -eq "vmms") -and
+            ($_.SystemCreationClassName -eq "Msvm_ComputerSystem") -and
+            ($_.SystemName -eq $env:COMPUTERNAME)
+        }
+
+    If (!$Svc)
+    {
+        Write-Error $Strings.ERR_NO_VMM
+    }
+
+    return $Svc
+}
+
+### -----------------------------------
+### Get-VMSgx Cmdlet
+### -----------------------------------
+
+Function Get-VMSgx
+{
+    [CmdletBinding()]
+    Param
+    (
+        # Name of the VM to get the SGX settings for.
+        [Parameter(Mandatory=$True, ParameterSetName="ByName", Position=0)]
+        [ValidateNotNullOrEmpty()]
+        [String]$VmName,
+
+        # VM to get the SGX settings for.
+        [Parameter(Mandatory=$True, ParameterSetName="ByObject", Position=0)]
+        [ValidateNotNull()]
+        [Microsoft.HyperV.PowerShell.VirtualMachine]$Vm
+    )
+
+    Process
+    {
+        If (!$Vm)
+        {
+            $Vm = Get-VM $VmName
+        }
+
+        $Mem = Get-VMSgxMemorySettingData $Vm
+        Return [VirtualMachineSgx]::New($Mem)
+    }
+}
+
+### -----------------------------------
+### Set-VMSgx Cmdlet
+### -----------------------------------
+
+Function Set-VMSgx
+{
+    [CmdletBinding()]
+    Param
+    (
+        # Name of the VM to configure.
+        [Parameter(Mandatory=$True, ParameterSetName="ByName", Position=0)]
+        [ValidateNotNullOrEmpty()]
+        [String]$VmName,
+
+        # VM to configure.
+        [Parameter(Mandatory=$True, ParameterSetName="ByObject", Position=0)]
+        [ValidateNotNull()]
+        [Microsoft.HyperV.PowerShell.VirtualMachine]$Vm,
+
+        # Whether SGX is enabled in the VM.
+        [Parameter(Position=1)]
+        [Bool]$IsSgxEnabled = $False,
+
+        # Desired SGX EPC Memory Size for the VM (in MB).
+        [Parameter(Position=2)]
+        [UInt64]$SgxSize = 0,
+
+        # Desired default SGX Launch Control Mode for the VM.
+        [Parameter(Position=3)]
+        [String]$SgxLaunchControlDefault = [String]::Empty,
+
+        # Desired SGX Launch Control Mode for the VM.
+        [Parameter(Position=4)]
+        [UInt32]$SgxLaunchControlMode = 0,
+
+        # Ignore warnings.
+        [Parameter()]
+        [Switch]$Force = $False
```

Comment:
Switch is $false by default.

Reply:
I added the explicit default for parity with the parameters above to express intent, but OK; removed.